### PR TITLE
Ch 1976

### PR DESCRIPTION
**Pull request by katbailey**
Track all actions in Lift Web

### CODE REVIEW

File: acquia_lift_profiles/acquia_lift_profiles.install [hunk 1]
```diff
@@ -121,3 +121,60 @@ function acquia_lift_profiles_update_7001() {
     menu_rebuild();
   }
 }
+
+/**
+ * Ensure all defined visitor actions are sync'd to Lift Web.
+ */
+function acquia_lift_profiles_update_7002(&$sandbox) {
```
Comment (eshtadc) [inline acquia_lift_profiles/acquia_lift_profiles.install:128]: Neat - had to look this up to find out about multi-pass updates.  I had no idea that this was an option.  Definitely cool.

File: acquia_lift_profiles/acquia_lift_profiles.module [hunk 2]
```diff
@@ -288,19 +288,8 @@ function acquia_lift_profiles_personalize_agent_save($agent) {
...
-    foreach ($tracked_actions as $action) {
-      if (in_array($action, $actions)) {
-        $client_side_actions[] = $action;
-      }
-    }
-  }
-
+  // Add info about client-side actions to Drupal.settings so that our client-
+  // side action listener can process them.
   if (!empty($client_side_actions)) {
```
Comment (eshtadc) [inline acquia_lift_profiles/acquia_lift_profiles.module]: Isn't $client_side_actions always empty here now that the above was deleted - or maybe this is supposed to check $actions?

File: acquia_lift_profiles/acquia_lift_profiles.module [hunk 3]
```diff
@@ -347,104 +336,33 @@ function acquia_lift_profiles_form_node_type_form_alter(&$form, &$form_state) {
...
-    $actions = visitor_actions_get_actions();
-    // Bail if called with a non-existent action name.
-    if (!isset($actions[$action_name])) {
-      drupal_set_message(t('Cannot save the action @action to Acquia Lift Profiles as it does not exist.', array('@action' => $action_name)), 'error');
-      return;
-    }
-    $label = !empty($actions[$action_name]['label']) ? $actions[$action_name]['label'] : $action_name;
-  }
-
-  // Visitor Actions module defaults the label to 'unlabeled', which is dumb, and
```
Comment (eshtadc) [inline acquia_lift_profiles/acquia_lift_profiles.module:432]: Yay for removing something "dumb" :-)

File: acquia_lift_profiles/acquia_lift_profiles.module [hunk 4]
```diff
@@ -347,104 +336,33 @@ function acquia_lift_profiles_form_node_type_form_alter(&$form, &$form_state) {
...
-/**
  * Calls the Acquia Lift Profiles API to create/update an event corresponding to a Visitor Actions action.
  *
  * @param $action_name
  *   The name of the action to save to Acquia Lift Profiles.
- * @param &$tracked_ations
- *   The array of actions being tracked in Lift Web, passed by reference so that it can
- *   be added to when this action is sync'd successfully.
  * @param string $label
  *   The human-readable name of the action. If not provided it will be looked up.
```
Comment (eshtadc) [inline acquia_lift_profiles/acquia_lift_profiles.module:344]: Should add $errors passthrough to the docblock

File: acquia_lift_profiles/acquia_lift_profiles.module [hunk 5]
```diff
@@ -473,11 +387,25 @@ function acquia_lift_profiles_delete_action($action_name, &$tracked_actions) {
...
-    acquia_lift_profiles_delete_action($action['machine_name'], $acquia_lift_profiles_tracked_actions);
-  }
-  variable_set('acquia_lift_profiles_tracked_actions', $acquia_lift_profiles_tracked_actions);
+  acquia_lift_profiles_delete_action($action['machine_name']);
+}
+
+/**
+ * Returns the human-readable name to use for an action.
+ *
+ * Visitor Actions module defaults the label to 'unlabeled', which is dumb, and
```
Comment (eshtadc) [inline acquia_lift_profiles/acquia_lift_profiles.module:398]: Oh.... "dumb" remains but on it's own, ha.

File: acquia_lift_profiles/acquia_lift_profiles.module (hunk 2 above)
Comment (katbailey) [inline acquia_lift_profiles/acquia_lift_profiles.module]: Doh! Nice catch - fixed

File: acquia_lift_profiles/acquia_lift_profiles.module (hunk 4 above)
Comment (katbailey) [inline acquia_lift_profiles/acquia_lift_profiles.module:344]: Done

**Pull request closed without merging**